BuRouteMobile/Login.tsx

```diff
@@ -0,0 +1,232 @@
+//import CheckBox from '@react-native-community/checkbox';
+import { useNavigation, useFocusEffect } from '@react-navigation/native';
+
+const Login = ({ navigation }) => {
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const [error, setError] = useState('');
+    const [success, setSuccess] = useState(''); 
+    const [loading, setLoading] = useState(false);
+    //const [rememberMe, setRememberMe] = useState(false);
+    //const navigation = useNavigation();
+
+    useFocusEffect(
+        React.useCallback(() => {
+            setUsername('');
+            setPassword('');
+            setError('');
+            setSuccess('');
+            setLoading(false);
+        }, [])
+    );
+
+    const handleLogin = async () => {
+        try {
+            if (!username || !password) {
+                setError('Please fill in all fields!');
+                setTimeout(() => setError(''), 2000);
+                return;
+            }
+            setLoading(true);
+            const response = await fetch('http://10.0.2.2:8000/database_search/login/', {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify({
+                    username,
+                    password
+                }),
+            });
+            //console.log("Request data:", JSON.stringify({ username, password }));
+            if (!response.ok) {
+                const errorResponse = await response.text();  // Get text response to understand the backend error
+                console.log("Login failed with response:", errorResponse);
+                setError("Invalid username or password! Please try again.");
+                setTimeout(() => setError(''), 2000); // Clear the error message after 2 seconds
+                return;
+            }
+            //assuming django server returns user data with successful login
+            const userData = await response.json();
+            setSuccess('Login successful! Logging you in...');
+            setTimeout(() => {
+                navigation.navigate('WikidataSearch', { userData }); // Navigate after showing success message
+                }, 2000); //navigate to WikidataSearch page after login (endpoint can change)
+        }
+        catch (error) {
+            //any unexpected error
+            setError(error.message);
+            console.log(error.message);
+            setTimeout(() => setError(''), 2000);
+        }
+        finally {
+            setLoading(false);
+        }
+    };
+
+    const handleSignUp = () => {
+        //navigate to sign up page
+        navigation.navigate('Signup');
+    };
+
+    const handleLoginWithGoogle = () => {
+        // NOT IMPLEMENTED YET
+        //navigate to WikidataSearch page after login (endpoint can change)
+        navigation.navigate('WikidataSearch');
```

Comment:
it should change in the actual implementation, it is navigating directly to the search page

Reply:
it was my fault btw, since i have not implemented google api usage for login yet, i was using this button for shortcut to search page. i forgot to change that.

Reply:
Okey, it's good you mentioned that. I'll try to implement that tonight, will inform you.